www/search.py

```diff
@@ -57,11 +58,45 @@ def GetTags(db, fid):
 		res.append(row[0])
 	return res
 
+def GetMapKeywordData(db):
+	rows = db.query("""SELECT fk.keyword,
+	COUNT(DISTINCT fk.fid) AS files, 
+	COUNT(DISTINCT f.name_without_version) AS maps 
+	FROM file_keyword fk 
+	INNER JOIN file f ON (fk.fid=f.fid) 
+	WHERE f.cid=2 GROUP BY fk.keyword ORDER BY fk.keyword ASC""")
+	res = []
+	for row in rows:
+		res.append(dict(row))
+	
+	return res
+
+def PrintMapKeywordList():
+	cfg = upqconfig.UpqConfig()
+	db = upqdb.UpqDB(cfg.db['url'], cfg.db['debug'])
+	rows = db.query("""SELECT CONCAT(f.name_without_version,COALESCE(CONCAT(';',GROUP_CONCAT(DISTINCT fk.keyword ORDER BY fk.keyword ASC SEPARATOR ';')),'')) AS line 
```

Comment:
not sure if i understood this function correctly, but doesn't this return the same as the contents on https://springrts.com/wiki/MapKeywords ?

if so: why? whats the purpose of this?

Reply:
The wiki data may be stale, weeks go by and new maps were added. That method can be used to get the fresh list of maps and keywords (some of which are added automatically) before making changes.